packages/@aws-cdk/aws-codedeploy/lib/lambda/custom-deployment-config.ts

```diff
@@ -0,0 +1,119 @@
+
+  /**
+   * The interval, in number of minutes:
+   * - For LINEAR, how frequently additional traffic is shifted
+   * - For CANARY, how long to shift traffic before the full deployment
+   */
+  readonly interval: string;
+}
+
+/**
+ * A custom Deployment Configuration for a Lambda Deployment Group.
+ *
+ */
+export class CustomLambdaDeploymentConfig extends Construct implements ILambdaDeploymentConfig {
+
+  /**
+   * The name of the deployment config
+   */
+  public readonly deploymentConfigName: string;
+
+  /**
+   * The arn of the deployment config
+   */
+  public readonly deploymentConfigArn: string;
+
+  public constructor(scope: Construct, id: string, props: CustomLambdaDeploymentConfigProps) {
+    super(scope, id);
+    const deploymentType = 'TimeBased' + props.type.toString();
+    let routingConfig;
+    let customDeploymentConfigName;
+    if (props.type == CustomLambdaDeploymentConfigType.CANARY) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedCanary: {
+          canaryInterval: props.interval,
+          canaryPercentage: props.percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaCanary' + props.percentage.toString() + 'Percent' + props.interval.toString() + 'Minutes';
+    } else if (props.type == CustomLambdaDeploymentConfigType.LINEAR) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedLinear: {
+          linearInterval: props.interval,
+          linearPercentage: props.percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaLinear' + props.percentage.toString() + 'Percent' + props.interval.toString() + 'Minutes';
+    } else {
+      throw new Error('Deployment config type not recognized, only CANARY and LINEAR are supported');
+    }
+
+    // What the config is called in the AWS console
+    const resourceName = id + customDeploymentConfigName;
+
+    // The AWS Custom Resource that calls CodeDeploy to create and delete the resource
+    new AwsCustomResource(this, customDeploymentConfigName + 'DeploymentConfig', {
+      onCreate: {
+        service: 'CodeDeploy',
+        action: 'createDeploymentConfig',
+        parameters: {
+          deploymentConfigName: resourceName,
+          computePlatform: 'Lambda',
+          trafficRoutingConfig: routingConfig,
+        },
+        physicalResourceId: PhysicalResourceId.of(resourceName),
+      },
+      onDelete: {
+        service: 'CodeDeploy',
+        action: 'deleteDeploymentConfig',
+        parameters: {
+          deploymentConfigName: resourceName,
+        },
+      },
+      policy: AwsCustomResourcePolicy.fromSdkCalls({
+        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
+      }),
+    });
+    this.deploymentConfigName = resourceName;
+    this.deploymentConfigArn = Fn.join(
+      ':',
+      ['arn', Stack.of(scope).partition, 'codedeploy', Stack.of(scope).region, Stack.of(scope).account, 'deploymentconfig', resourceName]);
```

Comment:
No need for any of this, we have a method `arnForDeploymentConfig` in `utils.ts`.

Reply:
Fixed